semgrep/semgrep/target_manager.py

```diff
@@ -188,12 +189,21 @@ def expand_targets(
         return expanded
 
     @staticmethod
-    def match_glob(path: Path, globs: List[str]) -> bool:
+    def preprocess_path_patterns(patterns: List[str]) -> List[str]:
```

Comment:
@brendongo 
Because calling `wcmatch`'s `PurePath.match` for each subpaths is too slow, I switched to this approach.
Let me know what do you think.
It looks a bit hacky, but I believe we can use unit tests to ensure it works as expected.

Reply:
Yeah looks good to me!